### PR TITLE
Implement Local UE Generation Handler Due to External Kafka Issues

### PR DESCRIPTION
**Pull request by tanzim10**
### Overview
This PR introduces a local User Equipment (UE) generation handler to replace the current API service, which is failing due to unresolved Kafka connectivity issues. The new local handler ensures continued functionality of UE simulations without external dependencies.

### Changes Made
- **Local UE Generation Classes**: Developed custom classes that handle the generation of UE data locally.
- **Mobility Data Generation**: This function generates UE mobility data by converting XY coordinates to longitude and latitude, organizing the data into a pandas DataFrame for each simulation tick across batches.
- **Data Output**: Implemented functionality to plot simulation results and save the data directly to CSV format for further analysis.

### Justification
The external API service for UE generation was reliant on Kafka streams, which are currently facing connectivity issues. To maintain productivity and ensure the reliability of our simulations, switching to a local generation method was deemed necessary.

### CODE REVIEW

File: radp/digital_twin/mobility/ue_generation_handler.py [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+        all_batches = []
+        for batch_no in range(self.num_batches):
+            batch = next(ue_generator.generate())
+            all_batches.append(batch)
+        print("Batches of Data: " , all_batches)
+        return all_batches
+
+
+
+    def mobility_data_generation(self) -> pd.DataFrame:
```
Comment (paulvarkey) [inline radp/digital_twin/mobility/ue_generation_handler.py:117]: this method and `generate_ue_data` can be replaced with the original `_mobility_data_generation`, which can actually be moved to this directory (as you are correctly doing in this PR), into `ue_tracks.py`, and we may rename it as `generate_as_lon_lat_points`, and then fix the call-site.

File: radp/digital_twin/mobility/ue_generation_handler.py [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+            all_dataframes.append(pd.concat(batch_dataframe_list, ignore_index=True))
+
+            if batch_no + 1 >= self.num_batches:  # Check if the specified number of batches has been reached
+                break
+
+        # Concatenate all batches into a single DataFrame and return
+        return pd.concat(all_dataframes, ignore_index=True)
+
+
+    def plot_ue_tracks(self,csv_file) -> None:
```
Comment (paulvarkey) [inline radp/digital_twin/mobility/ue_generation_handler.py:176]: make this a static method inside ue_tracks that takes a UETracksGenerator object and simulates and plots it. The whole simulation can be done and the whole data can be loaded into memory and plotted (as is being done here) OR, if possible, just enumerate through generate_as_lon_lat_points and plot one by one.

File: radp/digital_twin/mobility/ue_generation_handler.py [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+            plt.ylabel('Latitude')
+            plt.legend()
+
+            # Display the plot
+            plt.show()
+
+            # Update start_idx for the next batch
+            start_idx = end_idx
+
+    def save_data_to_csv(self, filename: str) -> None:
```
Comment (paulvarkey) [inline radp/digital_twin/mobility/ue_generation_handler.py:246]: you may delete this (in future, you may provide such a utility if we need to build in more business logic around this export)

**Review by paulvarkey — CHANGES_REQUESTED** [review]
This PR may also have a notebook that uses the API improvements here.

File: radp/digital_twin/mobility/ue_generation_handler.py (hunk 1 above)
Comment (tanzim10) [reply radp/digital_twin/mobility/ue_generation_handler.py:117]: Updated the ue_tracks by adding _mobility_data_generation from services/ue_tracks_generation/u_tracks_generation_driver.py file to ue_tracks.py. 
Renamed it to generate_as_lon_lat_points.

Will delete the reduntant ue_generation_handler.py after plot functions is fixed.

File: radp/digital_twin/mobility/ue_tracks_params.py [hunk 4]
```diff
@@ -0,0 +1,174 @@
+from typing import Dict
+
+from radp.common import constants
+from radp.digital_twin.mobility.ue_tracks import MobilityClass
+from services.ue_tracks_generation.ue_tracks_generation_helper import (
+    UETracksGenerationHelper,
+)
+
+
+class UETracksGenerationParams:
```
Comment (paulvarkey) [inline radp/digital_twin/mobility/ue_tracks_params.py:10]: add a unit test

**Review by paulvarkey — APPROVED** [review]
Approve -- please add `test_ue_tracks_params.py`

File: radp/digital_twin/mobility/ue_tracks_params.py (hunk 4 above)
Comment (tanzim10) [reply radp/digital_twin/mobility/ue_tracks_params.py:10]: Added the Unit Test Cases for the ue_tracks_params.
All of the changes are made.

Comment (tanzim10) [reply radp/digital_twin/mobility/ue_tracks_params.py:10]: You check the second last commit to see the changes made in mobility/test module. The last commit was done to format all of the codes using Black.

**Comment by paulvarkey** [comment]
Closing since done in #11

**Pull request closed without merging**